### PR TITLE
Add test helper to control eBPF build mode in tests

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Switches eBPF build mode control from outside of the test binary via environment variables, to a per-test or per-suite control.

- Adds the `ebpftest.TestBuildMode`/`ebpftest.TestBuildModes` and `ebpftest.SupportedBuildModes` functions for usage in tests.
- Converts existing tests to use the helpers and `testify/suite` where appropriate.
- Increases test timeouts because it must encompass the entire run, rather than just one ebpf build mode.
- Fixes OOM Kill test to be predictable and to work on all kernels by using `dd` with swap off.

### Motivation

1. Reduce waste by not running tests multiple times which do not vary based on eBPF build mode.
2. Add more control at the test-level as to which eBPF build modes to test, rather than the package-level.
3. Increase visibility of which eBPF build mode is being used in a particular test.

### Additional Notes

It is possible I missed some tests because this branch has existed for a bit, and I had to rebase many times. If you notice some tests that need to be updated, please let me know.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by ISauve — APPROVED** [review]
✔️ for the WKIT file

File: pkg/ebpf/ebpftest/buildmode_linux.go
Comment (guyarb) [inline pkg/ebpf/ebpftest/buildmode_linux.go:1]: I think adding `test` build tag is suitable here

Comment (guyarb) [reply pkg/ebpf/ebpftest/buildmode_linux.go:1]: and `linux` if the name of the file contains `linux`

File: pkg/ebpf/ebpftest/buildmode_windows.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
```
Comment (guyarb) [inline pkg/ebpf/ebpftest/buildmode_windows.go:4]: windows build tag?

Comment (brycekahle) [reply pkg/ebpf/ebpftest/buildmode_windows.go:4]: Unnecessary, because of Go naming conventions.

File: pkg/network/tracer/tracer_linux_test.go [hunk 2]
```diff
@@ -50,13 +50,8 @@ import (
...
 
-var kv470 kernel.Version = kernel.VersionCode(4, 7, 0)
-var kv kernel.Version
-
-func setKernelVersion() (err error) {
-	kv, err = kernel.HostVersion()
-	return
-}
+var kv470 = kernel.VersionCode(4, 7, 0)
+var kv = kernel.MustHostVersion()
```
Comment (guyarb) [inline pkg/network/tracer/tracer_linux_test.go:54]: back then @hmahmood was against not reporting the error

File: pkg/ebpf/ebpftest/buildmode_windows.go (hunk 1 above)
Comment (brycekahle) [reply pkg/ebpf/ebpftest/buildmode_windows.go:4]: > If a file's name, after stripping the extension and a possible _test suffix, matches any of the following patterns:
> `*_GOOS`
> `*_GOARCH`
> `*_GOOS_GOARCH`
> (example: source_windows_amd64.go) where GOOS and GOARCH represent any known operating system and architecture values respectively, then the file is considered to have an implicit build constraint requiring those terms (in addition to any explicit constraints in the file).

File: pkg/network/tracer/tracer_linux_test.go (hunk 2 above)
Comment (brycekahle) [reply pkg/network/tracer/tracer_linux_test.go:54]: In production code, I wouldn't panic, but this is in tests and will happen immediately.

**Review by guyarb — APPROVED** [review]

File: pkg/ebpf/ebpftest/buildmode_windows.go (hunk 1 above)
Comment (guyarb) [reply pkg/ebpf/ebpftest/buildmode_windows.go:4]: super useful tip, thanks

File: pkg/network/tracer/tracer_linux_test.go (hunk 2 above)
Comment (hmahmood) [reply pkg/network/tracer/tracer_linux_test.go:54]: `panic` works too :-)

**Review by safchain — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Review by Pythyu — APPROVED** [review]
LGTM !

**Review by L3n41c — APPROVED** [review]

File: pkg/ebpf/ebpftest/buildmode.go [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import (
+	"fmt"
+)
+
+// TODO I don't love fentry as a buildmode here...
+var (
+	Prebuilt        BuildMode
+	RuntimeCompiled BuildMode
+	CORE            BuildMode
+	Fentry          BuildMode
```
Comment (sunhay) [inline pkg/ebpf/ebpftest/buildmode.go:17]: Huh. Why is fentry it's own buildmode?

**Review by sunhay — APPROVED** [review]

**Pull request merged**